library/core/src/slice/iter.rs

```diff
@@ -2582,7 +2642,22 @@ impl<'a, T> DoubleEndedIterator for RChunksMut<'a, T> {
             let remainder = self.v.len() % self.chunk_size;
             let sz = if remainder != 0 { remainder } else { self.chunk_size };
             let tmp = mem::replace(&mut self.v, &mut []);
-            let (head, tail) = tmp.split_at_mut(sz);
+            // SAFETY: split_at_mut_unchecked requires the argument be less than
+            // or equal to the length. This is guaranteed, but subtle: `chunksz`
+            // will always either be `tmp_or_v.len() % self.chunk_size` (where
+            // `tmp_or_v` is the slice that at the time was `self.v` but now is
```

Comment:
Maybe it would be less confusing to just give `len` a binding and talk about that, i.e. `let len = ...`?

Reply:
I'm not exactly sure how that helps. What would I bind `len` to? Also, the correctness depends on what computation produced these.

Reply:
I ended up just referencing another safety comment, which avoids most of the `tmp_or_v` complexity. The `tmp` swap out that necessitated `tmp_or_v` that this does is only for the borrow checker, and doesn't really make it harder to understand the safety of the indexing, but was making it much more confusing to talk about.

Hopefully this is better.